### DISCUSSION TITLE
Use case for not charging battery but exporting all excess energy

### DISCUSSION THREAD

**Discussion by Kratax** [mletenay/home-assistant-goodwe-inverter#193] [Q&A]
Hi,

I would like to set up (if possible) in HA the following scenario having the experimental goodwe implementation:

For a set time (e.g. morning hours) - set the invertor in a way that the whole excess power is exported rather than stored in battery (i.e. following these priorities: the first is home use, the second export to grid; battery is only discharged if needed up to DOD, i.e. not charged in this mode).

Is it possible? Which mode and how to use?

Thanks a lot for superb work on this implementation!

Kratax

**Comment by Kratax** [comment]
Anybody can help with this? Thanks a lot!

**Comment by lampra1** [comment]
I would try setting Eco mode discharge to 1%

**Reply by ahammas** [reply]
This is how I solved it as well. Only issue is that if house consumes more energy than what PV can provide you'll end up buying energy so I set my inverter to discharge 15%. The ideal way of working would be that the battery can be set at maximum state of charge (say 30%) until midday (or cheap hours) and allow discharging if needed. Then, at a certain time or even better when it's cheap hours the battery charges and vice versa. Anybody figured out how to do this?
